### PR TITLE
ADBDEV-6389: Fix CTAS from prepared statement which contains modifying DML

### PR DESCRIPTION
**Pull request by RekGRpth**
Fix CTAS from prepared statement which contains modifying DML

The commit bad6ceb replaced the isCTAS boolean field in the Query structure
with a parentStmtType uint8, but forgot to change its read/write from
READ_BOOL_FIELD/WRITE_BOOL_FIELD to READ_UINT_FIELD/WRITE_UINT_FIELD and
READ_UINT8_FIELD/WRITE_UINT8_FIELD. This patch fixes that bug as well.

The commit 93abe74 added conditions on modifying DMLs, but left the previous
assert, which does not allow CTAS. At the same time, when executing CTAS from
prepared statements which containes modifying DML, this assert should allow
such case. This patch solves the problem.

The commit 19cd1cf introduced an incorrect field reading order in the
_readQuery function. This patch fixes that issue as well.

### CODE REVIEW

File: src/backend/nodes/outfast.c [hunk 1]
```diff
@@ -528,7 +528,7 @@ _outQuery(StringInfo str, Query *node)
 	WRITE_NODE_FIELD(rowMarks);
 	WRITE_NODE_FIELD(setOperations);
 	WRITE_NODE_FIELD(constraintDeps);
-	WRITE_BOOL_FIELD(parentStmtType);
+	WRITE_ENUM_FIELD(parentStmtType, ParentStmtType);
```
Comment (red1452) [inline src/backend/nodes/outfast.c]: `ParentStmtType` is uint8. I think, it is incorrect macro for this type. You can change `ParentStmtType` or create macro for uint8.

File: src/backend/nodes/readfast.c [hunk 2]
```diff
@@ -269,7 +269,7 @@ _readQuery(void)
 	READ_NODE_FIELD(rowMarks);
 	READ_NODE_FIELD(setOperations);
 	READ_NODE_FIELD(constraintDeps);
-	READ_BOOL_FIELD(parentStmtType);
+	READ_ENUM_FIELD(parentStmtType, ParentStmtType);
```
Comment (red1452) [inline src/backend/nodes/readfast.c]: Add assert, as it was done above.

File: src/backend/nodes/outfast.c (hunk 1 above)
Comment (RekGRpth) [reply src/backend/nodes/outfast.c]: fixed

File: src/backend/nodes/readfast.c (hunk 2 above)
Comment (RekGRpth) [reply src/backend/nodes/readfast.c]: fixed

**Comment by red1452** [comment]
Remarks:

1) prepared statement WHICH contains
2) such case

**Comment by RekGRpth** [comment]
> Remarks:
> 
>     1. prepared statement WHICH contains
> 
>     2. such case

applied

**Review by red1452 — APPROVED** [review]

**Review by bimboterminator1 — APPROVED** [review]

**Comment by RekGRpth** [comment]
bender build

**Comment by swork12** [comment]
bender build

**Comment by swork12** [comment]
bender build

**Pull request merged**